### ISSUE TITLE
How to get "Retry-After" value inside error_message or errorhandler ?

### ISSUE THREAD

**Issue by cocxtrash** [alisaifee/flask-limiter#276]
Instead of default n per x error message, i want to show user something more precise, like "Retry After 2 minutes 34 seconds". 

And for that i need the "Retry-After" delta-seconds value.

Please help. Thank You.

**Comment by alisaifee** [comment]
This is unfortunately not possible at the moment since the context for getting the retry after value is lost when the error handler or error message are evaluated. This does seem like a useful feature to have so I'll try to add it in the next few days.

**Comment by gruvw** [comment]
I would really like to have this feature too! This is exactly what I was looking for... It would be nice to have access to the Retry-After value before sending the response back to the client. That way we could use it either in the `error_message` argument or even in the `app.errorhandler(429)` function in order to display it back to the client on top of the message bound to the current limiter.

**Comment by cocxtrash** [comment]
Hi, Any progress? Commenting so stale bot don't eat it.

Also with **RATELIMIT_HEADER_RETRY_AFTER** value, if possible please include all **RATELIMIT_HEADER_*** values.

Thank You.

**Comment by ongkbk** [comment]
> This is unfortunately not possible at the moment since the context for getting the retry after value is lost when the error handler or error message are evaluated. This does seem like a useful feature to have so I'll try to add it in the next few days.

Has there been any update on this?

**Comment by giigaby** [comment]
Do we have any way to return the response headers like  Retry-After, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset? 

With the after_request I am not getting those headers :/ 

```
@app.after_request
def after_request_func(response):
    print(response.status)
    print(response.headers)
    return response
```
![image](https://user-images.githubusercontent.com/49292595/148651903-ba6a4111-315c-476c-bed6-f948491ad617.png)

![image](https://user-images.githubusercontent.com/49292595/148651851-62ce8a3d-4a7b-4fba-b1bc-62a66eb88f18.png)

**Comment by alisaifee** [comment]
My apologies for the delay in getting back to this request.
I've added a new property `header_limit` on the `Limiter` instance which can be used to extract details for the most relevant rate limit for the given request. This is in master and not yet released - but there are some details in the documentation [here]( https://flask-limiter.readthedocs.io/en/latest/api.html#flask_limiter.Limiter.header_limit)

A very contrived example that accesses the details in both the `@after_request` and `@errorhandler` hooks:

```python
from flask import Flask, jsonify, make_response

from flask_limiter import Limiter

app = Flask(__name__)
limiter = Limiter(
    app,
    lambda: "test",
    headers_enabled=True,
    default_limits=["1/10 seconds"],
)


@app.route("/")
@limiter.limit("1/second")
@limiter.limit("60/minute")
@limiter.limit("2/day")
def test():
    return jsonify(
        remaining=limiter.header_limit.remaining,
        key=limiter.header_limit.key,
        reset_at=limiter.header_limit.reset_at,
    )


@app.after_request
def add_details(response):
    response.headers["X-Count-Remaining"] = limiter.header_limit.remaining
    response.headers["X-RateLimit-Reset-Timestamp"] = limiter.header_limit.reset_at

    return response


@app.errorhandler(429)
def error_handler(e):
    response = e.get_response()
    response.headers["X-Teapot"] = True

    return response


app.run()

```

**Comment by alisaifee** [comment]
Minor modification - I had previously suggested adding a `header_limit` property - but I've changed that to `current_limit` - snippet above updated to reflect that change.

This will be available in the next release.

**Comment by alisaifee** [comment]
This is now available in `2.1.x`
